test/integration/mesh_uri.cc

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2023 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include <gz/msgs/entity_factory.pb.h>
+
+#include <gz/common/Console.hh>
+#include <gz/common/Util.hh>
+#include <gz/math/Pose3.hh>
+#include <gz/transport/Node.hh>
+#include <gz/utils/ExtraTestMacros.hh>
+
+#include "gz/sim/components/Model.hh"
+#include "gz/sim/components/Name.hh"
+#include "gz/sim/components/Pose.hh"
+#include "gz/sim/Server.hh"
+#include "gz/sim/SystemLoader.hh"
+#include "test_config.hh"
+
+#include "../helpers/Relay.hh"
+#include "../helpers/EnvTestFixture.hh"
+
+using namespace gz;
+using namespace sim;
+
+std::mutex mutex;
+int g_count = 0;
+unsigned char *g_imageBuffer = nullptr;
+msgs::Image g_image;
+
+/////////////////////////////////////////////////
+void cameraCb(const msgs::Image & _msg)
+{
+  ASSERT_EQ(msgs::PixelFormatType::RGB_INT8,
+      _msg.pixel_format_type());
+
+  std::lock_guard<std::mutex> lock(mutex);
+  g_image = _msg;
+  g_count++;
+}
+
+std::string meshModelStr(bool _static = false)
+{
+  // SDF string consisting of a box mesh. Note that "unit_box"
+  // is a mesh that comes pre-registered with MeshManager so
+  // it should be able to load this as a mesh.
+  // Similarly the user can create a common::Mesh object and add that
+  // to the MeshManager by calling MeshManager::Instance()->AddMesh;
+  return std::string("<?xml version=\"1.0\" ?>") +
+      "<sdf version='1.6'>" +
+      "<model name='spawned_model'>" +
+      "<link name='link'>" +
+      "<visual name='visual'>" +
+      "<geometry><mesh><uri>unit_box</uri></mesh></geometry>" +
```

Comment:
Should we use some sort of url scheme for this, like, `memory://unit_box` or similar to avoid confusion? Maybe something to bring up to the team.

Reply:
they other idea I had before is to add a `<mesh><name>` tag. A new scheme like `memory://` (or `name://`>?) also works.

Reply:
Per discussion in the simulation meeting, `name://` has been selected. Do you mind adding some documentation, maybe in https://github.com/gazebosim/gz-sim/blob/gz-sim7/tutorials/resources.md?

Reply:
Updated to parse URI with `name://` scheme. [b5428a1](https://github.com/gazebosim/gz-sim/pull/2007/commits/b5428a122ee085d0929922949331348578a97b46). 
Added doc in resources tutorial. [1bcf7ac](https://github.com/gazebosim/gz-sim/pull/2007/commits/1bcf7ac4a26a19fa719fa46ed48f2451c989b3a3)